### PR TITLE
tests: update_config bugfix

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

File: cli/tests/integrations/helpers/common.py [hunk 1]
```diff
@@ -204,9 +204,9 @@ def update_config(name, value, env=None):
     # we need to also restore the token if this occurs
     token = None
     if name == "core.dcos_url":
-        returncode, token_val, _ = exec_command(
+        returncode_, token_val, _ = exec_command(
             ['dcos', 'config', 'show', "core.dcos_acs_token"], env)
-        if returncode == 0:
+        if returncode_ == 0:
```
Comment (tamarrow) [inline cli/tests/integrations/helpers/common.py]: Notice that we we're overriding `returncode` [here](https://github.com/dcos/dcos-cli/pull/970/files#diff-3486a2e5d4e0e340bd57f8d6279a2495R200) which gets used [here](https://github.com/dcos/dcos-cli/pull/970/files#diff-3486a2e5d4e0e340bd57f8d6279a2495R214)

**Comment by tamarrow** [comment]
cc @msabramo

File: cli/tests/integrations/helpers/common.py (hunk 1 above)
Comment (msabramo) [inline cli/tests/integrations/helpers/common.py]: Oh I see. It was clobbering the previous value of `returncode` so if you don't have a token, it thinks you have no config and doesn't try to update it.
👍

Comment (tamarrow) [inline cli/tests/integrations/helpers/common.py]: exactly :)

**Comment by msabramo** [comment]
👍  as this got more integration tests passing for me.

```
=============================================================================================== test session starts ================================================================================================
platform darwin -- Python 3.5.3, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- /Users/abramowi/dev/git-repos/dcos-cli/cli/.tox/py35-integration/bin/python3.5
rootdir: /Users/abramowi/dev/git-repos/dcos-cli/cli, inifile:
plugins: teamcity-messages-1.21, cov-2.4.0
collected 324 items

tests/integrations/test_auth.py::test_info PASSED
tests/integrations/test_auth.py::test_version PASSED
tests/integrations/test_auth.py::test_logout_no_token PASSED
tests/integrations/test_auth.py::test_logout_with_token PASSED
tests/integrations/test_config.py::test_info PASSED
tests/integrations/test_config.py::test_version PASSED
tests/integrations/test_config.py::test_get_existing_string_property PASSED
tests/integrations/test_config.py::test_get_existing_boolean_property PASSED
tests/integrations/test_config.py::test_get_existing_number_property PASSED
tests/integrations/test_config.py::test_get_missing_property PASSED
tests/integrations/test_config.py::test_dcos_url_without_scheme PASSED
tests/integrations/test_config.py::test_get_top_property PASSED
tests/integrations/test_config.py::test_set_package_sources_property PASSED
tests/integrations/test_config.py::test_set_core_email_property PASSED
tests/integrations/test_config.py::test_set_existing_string_property PASSED
tests/integrations/test_config.py::test_set_existing_boolean_property PASSED
tests/integrations/test_config.py::test_set_existing_number_property PASSED
tests/integrations/test_config.py::test_set_change_output PASSED
tests/integrations/test_config.py::test_set_same_output PASSED
tests/integrations/test_config.py::test_set_new_output PASSED
tests/integrations/test_config.py::test_set_nonexistent_subcommand PASSED
tests/integrations/test_config.py::test_set_when_extra_section PASSED
tests/integrations/test_config.py::test_unset_property PASSED
tests/integrations/test_config.py::test_unset_missing_property PASSED
tests/integrations/test_config.py::test_unset_output PASSED
tests/integrations/test_config.py::test_unset_top_property PASSED
tests/integrations/test_config.py::test_validate PASSED
tests/integrations/test_config.py::test_set_property_key PASSED
tests/integrations/test_config.py::test_set_missing_property PASSED
tests/integrations/test_config.py::test_set_core_property PASSED
tests/integrations/test_config.py::test_url_validation PASSED
tests/integrations/test_config.py::test_fail_url_validation PASSED
tests/integrations/test_config.py::test_bad_port_fail_url_validation PASSED
tests/integrations/test_config.py::test_dcos_url_env_var PASSED
tests/integrations/test_config.py::test_dcos_dcos_url_env_var PASSED
tests/integrations/test_config.py::test_timeout SKIPPED
tests/integrations/test_config.py::test_parse_error PASSED
tests/integrations/test_dcos.py::test_default FAILED
```

**Comment by msabramo** [comment]
A ha, I think `tests/integrations/test_dcos.py::test_default` is failing for me because I have the `tunnel-cli` subcommand installed.

```
$ ls -l ~/.dcos/subcommands
total 0
drwxr-xr-x  4 abramowi  staff  136 Mar 27 17:08 tunnel-cli/
```

Is there a way to disable subcommands?

**Comment by msabramo** [comment]
I don't see a way to disable custom subcommands; how about something like this (I can send a PR if you like):

```diff
--- a/cli/tests/data/dcos.toml
+++ b/cli/tests/data/dcos.toml
@@ -1,5 +1,6 @@
 [core]
-dcos_url = "http://dcos.snakeoil.mesosphere.com"
 ssl_verify = "false"
 timeout = 5
+dcos_url = "http://dcos.snakeoil.mesosphere.com"
 reporting = false
+disable_custom_subcommands = true
diff --git a/dcos/data/config-schema/core.json b/dcos/data/config-schema/core.json
index 9c8107f..7f09f96 100644
--- a/dcos/data/config-schema/core.json
+++ b/dcos/data/config-schema/core.json
@@ -19,6 +19,12 @@
             "type": "boolean",
             "default": false
         },
+        "disable_custom_subcommands": {
+            "description": "Whether to disable custom subcommands; useful for testing for example",
+            "title": "Disable custom subcommands",
+            "type": "boolean",
+            "default": false
+        },
         "mesos_master_url": {
             "description": "Mesos master URL. Must be set in format: \"http://host:port\"",
             "format": "uri",
diff --git a/dcos/subcommand.py b/dcos/subcommand.py
index b213f2f..7a0d701 100644
--- a/dcos/subcommand.py
+++ b/dcos/subcommand.py
@@ -95,8 +95,13 @@ def list_paths():
     """

     subcommands = []
-    for package in distributions():
-        subcommands += get_package_commands(package)
+
+    toml_config = config.get_config(True)
+    disable_custom_subcommands = config.get_config_val(
+        "core.disable_custom_subcommands", toml_config)
+    if not disable_custom_subcommands:
+        for package in distributions():
+            subcommands += get_package_commands(package)

     return subcommands
```

**Comment by tamarrow** [comment]
Not a big fan of options for testing only. But we're adding management for having several clusters, and subcommands will be able to be per cluster only.

**Comment by msabramo** [comment]
Yeah I thought that maybe your multi-cluster stuff (which I haven't looked at) might handle this better. :+1:

**Pull request merged**